main.py

```diff
@@ -29,21 +39,14 @@ class MainWindow(QMainWindow):
     def __init__(self):
         
         def collect_route(button):
```

Comment:
why do you define functions inside the __init__

Reply:
is it because of the reference to self?

Reply:
also why is all the logic in the __init__()

Reply:
All my logic was once in the init but I slowly moved everything into modules but this is the only function I haven't figured out how to refactor because I've not figured out how to pass the button and the button group as an argument. 

This is actually my main priority to fix at the moment as the function does too much (gets button id, creates route list and also assigns the list to the climb) and also isn't doing what I need it to which is just return a list of button IDs as they're clicked. 

The button is passed within line 73 of main.py but as soon as I added the button group as an argument to the collect_route function it stopped working as expected and I need the button group so I need to completely redesign how I collect the route from the user